theano/compile/mode.py

```diff
@@ -111,6 +111,8 @@ class AddDestroyHandler(gof.Optimizer):
 
     """
     def apply(self, fgraph):
+        if theano.config.cycle_detection == 'fast':
```

Comment:
This is what we had discussed previously with Pascal. When we have the ```fast```, we don't want to have the check for the ```outputguard```.  There is no failing test on this PR. All the tests have passed.